flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -161,10 +161,10 @@ public void disableCheckpointing() {
     /**
      * Checks whether checkpointing is enabled.
      *
-     * @return True if checkpointing is enables, false otherwise.
+     * @return True if checkpointing is enabled, false otherwise.
      */
     public boolean isCheckpointingEnabled() {
-        return checkpointInterval > 0;
+        return checkpointInterval >= MINIMAL_CHECKPOINT_TIME && checkpointTimeout < Long.MAX_VALUE;
```

Comment:
Factor this out into a method which can also be used by the `JobGraph` to keep the test criterion consistent.